src/client.ts

```diff
@@ -501,11 +501,15 @@ export class SubscriptionClient {
       this.closedByUser = false;
       this.eventEmitter.emit(this.reconnecting ? 'reconnecting' : 'connecting');
 
-      const payload: ConnectionParams = typeof this.connectionParams === 'function' ? this.connectionParams() : this.connectionParams;
-
-      // Send CONNECTION_INIT message, no need to wait for connection to success (reduce roundtrips)
-      this.sendMessage(undefined, MessageTypes.GQL_CONNECTION_INIT, payload);
-      this.flushUnsentMessagesQueue();
+      if (typeof this.connectionParams === 'function') {
```

Comment:
This line can remain the same. And then below we can just `Promise.resolve(payload).then(...)`.